xml/issue1203.xml

```diff
@@ -134,6 +143,58 @@ and only if <tt>Ostream</tt> is not an lvalue reference type and is derived from
 </ins></p>
 </blockquote>
 </blockquote>
+</blockquote>
+</discussion>
+
+<resolution>
+<p>This resolution is relative to N4750.</p>
+<p>
+Change <sref ref="[istream.rvalue]"/> as follows:
+</p>
+
+<blockquote><pre>
+template &lt;class <del>charT, class traits</del> <ins>Istream</ins>, class T&gt;
+  <del>basic_istream&lt;charT, traits&gt;&amp;</del> <ins>Istream&amp;&amp;</ins>
+  operator&gt;&gt;(<del>basic_istream&lt;charT, traits&gt;</del> <ins>Istream</ins>&amp;&amp; is, T&amp;&amp; x);
+</pre>
+<blockquote>
+<p>
+-1- <i>Effects:</i> Equivalent to:
+</p>
+<pre>
+<tt>is &gt;&gt; std::forward&lt;T&gt;(x)</tt>
+return <ins>std::move(</ins>is<ins>)</ins>;
+</pre>
+<p>-2- <i>Remarks:</i> This function shall not participate in overload
+resolution unless the expression <tt>is &gt;&gt; std::forward&lt;T&gt;(x)</tt> is
+well-formed<ins>, <tt>Istream</tt> is not an lvalue reference type, and
+<tt>Istream</tt> is derived from <tt>ios_base</tt></ins>.</p>
```

Comment:
Suggest "publicly and unambiguously derived from" to avoid weirdness like `tuple<stream, double>{} >> foo`. (And on 195)

Reply:
I agree that would be better; but that goes beyond a "rebase" which I tried to do in this change.

@Dani-Hub @mclow How do you suggest I proceed?

Reply:
Doesn't such weirdness break the expression well-formedness check already?

Reply:
@timsong-cpp No reason a user can't write:

```
std::tuple<stream> operator<<(std::tuple<stream>&, SomeUserTypeToGetThisThingInADL)
```

Reply:
Sheer insanity. All right.

Reply:
Rule of thumb: put ADL expression constraints last. You can never foresee all the insanity that ADL will find, and you don't want to pay the cost of overload resolution if it's avoidable.

Reply:
That's how I *implemented* it @CaseyCarter -- just didn't think about making the implementation order match the standard order.